controllers/node_kernel_clusterclaim.go

```diff
@@ -0,0 +1,112 @@
+package controllers
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	"github.com/kubernetes-sigs/kernel-module-management/internal/filter"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"open-cluster-management.io/api/cluster/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+//+kubebuilder:rbac:groups="core",resources=nodes,verbs=get;patch;list;watch
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=clusterclaims,resourceNames=kernel-versions.kmm.node.kubernetes.io,verbs=create;delete;get;list;patch;update;watch
+
+const clusterClaimName = "kernel-versions.kmm.node.kubernetes.io"
+
+type NodeKernelClusterClaimReconciler struct {
+	client client.Client
+}
+
+func NewNodeKernelClusterClaimReconciler(client client.Client) *NodeKernelClusterClaimReconciler {
+	return &NodeKernelClusterClaimReconciler{client: client}
+}
+
+func (r *NodeKernelClusterClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	nodeList := v1.NodeList{}
+
+	logger.Info("Listing all nodes")
+
+	if err := r.client.List(ctx, &nodeList); err != nil {
```

Comment:
do we want to add a check for Node status? working only on nodes with Ready status?

Reply:
If anything, I would filter out the unschedulable (tainted) nodes like you did in KMM. What are your thoughts on this? Its is reasonable to assume that those nodes will never need an OOT module?